### PR TITLE
Reset Password Page

### PR DESCRIPTION
**Pull request by jonathanjma**
- Add page/modal for user to reset their password after receiving the password reset email
- Modal also allows the user to enter their journal recovery phrase (and warns them what will happen if they do not)
- Add text to private journal login page to remind users to set up a recovery phrase in settings
- Also fixes z-index bug related to modals and a bug with the journal count API request

### CODE REVIEW

File: src/index.css [hunk 1]
```diff
@@ -85,11 +85,15 @@ label:hover {
...
 
 .transition-bg-500 {
     -webkit-transition: background-color 500ms linear;
     -ms-transition: background-color 500ms linear;
     transition: background-color 500ms linear;
 }
+
+.radial {
+    background: radial-gradient(50% 50% at 50% 50%, rgba(240, 207, 120, 0.30) 0%, rgba(255, 255, 255, 0.00) 100%), #F7EFD7;
+}
```
Comment (zachseidner1) [inline src/index.css:99]: thanks for adding this 😍

File: src/pages/Entries/EntryCard.tsx [hunk 2]
```diff
@@ -182,9 +180,14 @@ export default function EntryCard({
         id="delete-confirm-modal"
         title="Deleting happiness"
         // fix comment form
-        body={`You are deleting happiness for ${parseYYYYmmddFormat(
+        body={`You are deleting happiness for ${dateFromStr(
```
Comment (zachseidner1) [inline src/pages/Entries/EntryCard.tsx:183]: why did you change out the date function here? just curious

File: src/pages/ResetPassword/ResetPassword.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  const [triedToSubmit, setTriedToSubmit] = useState(false);
+  const [recover, setRecover] = useState(false);
+  const [recoveryPhrase, setRecoveryPhrase] = useState("");
+
+  const [error, setError] = useState("");
+
+  useEffect(() => {
+    // redirect to home if no token is provided
+    if (resetToken) {
+      window.HSOverlay.open(document.querySelector("#reset-pass-modal"));
```
Comment (zachseidner1) [inline src/pages/ResetPassword/ResetPassword.tsx:30]: I would wrap this in a try catch just to be safe, I've seen this function crash the code if it tries to load the modal before the page is loaded.

**Review by zachseidner1 — APPROVED** [review]
LGTM! I left a couple of comments I'd appreciate if you respond to them but nothing major, so LGTM.

File: src/pages/ResetPassword/ResetPassword.tsx (hunk 3 above)
Comment (jonathanjma) [reply src/pages/ResetPassword/ResetPassword.tsx:31]: added a check to make sure the modal plugin is loaded

File: src/pages/Entries/EntryCard.tsx (hunk 2 above)
Comment (jonathanjma) [reply src/pages/Entries/EntryCard.tsx:183]: i think i just did that to be consistent with the other change I made on the same file
